## Project: Russian Oligarch Problemzz

Problem:
I am a very rich Russian oligarch and alot of my assets are going to get sanctioned, including alot of my cash.
I need to buy property and expensive property, quickly, but i want to be descrete about it.

I want to buy a property NY but i need to figure out where to look.

I need to look for the following:

1. The neighborhood that has the most residential properties over 10m 
2. The average price of each property type

From there I will be able to quickly make an informed decision to quickly buy a property!!

Steps I will need to answer these two questions:
- Download the data and convert it to a list of lists
- Isolate all the properties that are single unit residential that are also not commercial
- Ensure prices are in float so i can take averages 
- Create a frequency table for each neighboorhood on properties over 10m
- Create a dictionary with the average price of each building class category
- Answer question

Step 1: Download the data and convert it to a list of lists

In [7]:
opened_file1 = open('nyc-rolling-sales.csv')
from csv import reader
read_file1 = reader(opened_file1)
nycdata = list(read_file1)

nycdata[0:5]

[['',
  'BOROUGH',
  'NEIGHBORHOOD',
  'BUILDING CLASS CATEGORY',
  'TAX CLASS AT PRESENT',
  'BLOCK',
  'LOT',
  'EASE-MENT',
  'BUILDING CLASS AT PRESENT',
  'ADDRESS',
  'APARTMENT NUMBER',
  'ZIP CODE',
  'RESIDENTIAL UNITS',
  'COMMERCIAL UNITS',
  'TOTAL UNITS',
  'LAND SQUARE FEET',
  'GROSS SQUARE FEET',
  'YEAR BUILT',
  'TAX CLASS AT TIME OF SALE',
  'BUILDING CLASS AT TIME OF SALE',
  'SALE PRICE',
  'SALE DATE'],
 ['4',
  '1',
  'ALPHABET CITY',
  '07 RENTALS - WALKUP APARTMENTS             ',
  '2A',
  '392',
  '6',
  ' ',
  'C2',
  '153 AVENUE B',
  ' ',
  '10009',
  '5',
  '0',
  '5',
  '1633',
  '6440',
  '1900',
  '2',
  'C2',
  '6625000',
  '2017-07-19 00:00:00'],
 ['5',
  '1',
  'ALPHABET CITY',
  '07 RENTALS - WALKUP APARTMENTS             ',
  '2',
  '399',
  '26',
  ' ',
  'C7',
  '234 EAST 4TH   STREET',
  ' ',
  '10009',
  '28',
  '3',
  '31',
  '4616',
  '18690',
  '1900',
  '2',
  'C7',
  ' -  ',
  '2016-12-14 00:00:00'],
 ['6',
  '1',
  'ALPHABET CITY',
  '07

Step 2: Isolate all the properties that are single unit residential that are also not commerical

In [43]:
singleresident = []
for row in nycdata[1:]:
    totalunit = row[-8]
    commercial = row[-9]
    residential = row[-10]
    if (totalunit == '1') and (commercial == '0') and (residential == '1'):
        singleresident.append(row)
        
len(singleresident)


34194

There are 34,000 properties that are single resident, now we need to see how many have a price

In [8]:
pricesinglerez = []
for row in singleresident:
    price = row[-2]
    if price != ' -  ':
        pricesinglerez.append(row)
        
len(pricesinglerez)

27982

Ok, 28,000 have prices on them. Again, this means we have 28,000 properties across NYC that are single residential properties.

Step 3: Ensure prices are in float so we can work with them

In [19]:
for row in pricesinglerez:
    price = int(row[-2])


    

Step 4: Create a frequency table of properties that are over 10m by neighborhood.

In [23]:
tenmil_prop = []

for row in pricesinglerez:
    price = len(row[-2])
    target = 8
    if price >= target:
        tenmil_prop.append(row)

len(tenmil_prop)
 

286

Ive done very sneaky here, was getting some level of errors so i figured to count by how many numbers were in price. 8 Digits is over 10m so i did length of the price.

In [27]:
hood_dic = {}

for row in tenmil_prop:
    hood = row[2]
    if hood in hood_dic:
        hood_dic[hood] += 1
    else:
        hood_dic[hood] = 1 
hood_dic



{'CHELSEA': 25,
 'CIVIC CENTER': 37,
 'CLINTON': 1,
 'EAST VILLAGE': 1,
 'FINANCIAL': 4,
 'FLATIRON': 24,
 'GRAMERCY': 3,
 'GREENWICH VILLAGE-CENTRAL': 6,
 'GREENWICH VILLAGE-WEST': 32,
 'LITTLE ITALY': 2,
 'MIDTOWN CBD': 25,
 'MIDTOWN EAST': 2,
 'MIDTOWN WEST': 13,
 'SOHO': 8,
 'TRIBECA': 29,
 'UPPER EAST SIDE (59-79)': 18,
 'UPPER EAST SIDE (79-96)': 17,
 'UPPER WEST SIDE (59-79)': 25,
 'UPPER WEST SIDE (79-96)': 5,
 'MORRISANIA/LONGWOOD': 1,
 'BROOKLYN HEIGHTS': 3,
 'DOWNTOWN-FULTON FERRY': 1,
 'DOWNTOWN-FULTON MALL': 1,
 'PARK SLOPE': 1,
 'TOTTENVILLE': 1,
 'TRAVIS': 1}

Sort from most to least

In [31]:
sorted_hood_dic = sorted(hood_dic.items() , key=lambda item: item[1], reverse=True)
sorted_hood_dic

[('CIVIC CENTER', 37),
 ('GREENWICH VILLAGE-WEST', 32),
 ('TRIBECA', 29),
 ('CHELSEA', 25),
 ('MIDTOWN CBD', 25),
 ('UPPER WEST SIDE (59-79)', 25),
 ('FLATIRON', 24),
 ('UPPER EAST SIDE (59-79)', 18),
 ('UPPER EAST SIDE (79-96)', 17),
 ('MIDTOWN WEST', 13),
 ('SOHO', 8),
 ('GREENWICH VILLAGE-CENTRAL', 6),
 ('UPPER WEST SIDE (79-96)', 5),
 ('FINANCIAL', 4),
 ('GRAMERCY', 3),
 ('BROOKLYN HEIGHTS', 3),
 ('LITTLE ITALY', 2),
 ('MIDTOWN EAST', 2),
 ('CLINTON', 1),
 ('EAST VILLAGE', 1),
 ('MORRISANIA/LONGWOOD', 1),
 ('DOWNTOWN-FULTON FERRY', 1),
 ('DOWNTOWN-FULTON MALL', 1),
 ('PARK SLOPE', 1),
 ('TOTTENVILLE', 1),
 ('TRAVIS', 1)]

Ok so looks like Civic center  has the most 10m apartments. Ill buy there.

Step 5: Find the average price for each unit type in Civic center, whatever is highest I will try to find something like that.

Create a frequency table for the types of units in Civic center

In [34]:
civic_center_types = {}

for row in tenmil_prop:
    hood = row[2]
    building = row[3]
    if building in civic_center_types:
        civic_center_types[building] += 1
    else:
        civic_center_types[building] = 1 
    
civic_center_types

{'13 CONDOS - ELEVATOR APARTMENTS            ': 252,
 '15 CONDOS - 2-10 UNIT RESIDENTIAL          ': 8,
 '04 TAX CLASS 1 CONDOS                      ': 1,
 '01 ONE FAMILY DWELLINGS                    ': 24,
 '26 OTHER HOTELS                            ': 1}

Ok, this is the break down of building types in civic center that are over 10m. Lets find the average price for each.

In [46]:
for building in civic_center_types:
    total = 0
    len_category = 0
    for row in tenmil_prop:
        build_type = row[3]
        if build_type == building:
            total_price = row[-2]
            total += float(total_price)
            len_category += 1
    average = total / len_category
    print(building, average)

13 CONDOS - ELEVATOR APARTMENTS             18811302.845238097
15 CONDOS - 2-10 UNIT RESIDENTIAL           14800733.375
04 TAX CLASS 1 CONDOS                       13600000.0
01 ONE FAMILY DWELLINGS                     17375570.75
26 OTHER HOTELS                             11700000.0


By the looks of it, the highest average price of Civic Center property types is 18.8 million, which is Condos with elevator apartments. 

## Conclusion and Findings:

Concluding Advice:

- Look to buy a Condo property with Elevator apartments in Civic Center.

Explanation:
- These are very common in civic center, there are over 252. This is the least sus way to spend the most money as its common in this area. Good luck!